Sources/MongoSwift/MongoConnectionString.swift

```diff
@@ -0,0 +1,107 @@
+/// Represents a MongoDB connection string.
+/// - SeeAlso: https://docs.mongodb.com/manual/reference/connection-string/
+public struct MongoConnectionString: Codable, LosslessStringConvertible {
+    /// Represents a connection string scheme.
+    public struct Scheme: LosslessStringConvertible, Equatable {
+        /// Indicates that this connection string uses the scheme `mongodb`.
+        public static let mongodb = Scheme(.mongodb)
+
+        /// Indicates that this connection string uses the scheme `mongodb+srv`.
+        public static let srv = Scheme(.srv)
+
+        /// Internal representation of a scheme.
+        private enum _Scheme: String {
+            case mongodb
+            case srv = "mongodb+srv"
+        }
+
+        private let _scheme: _Scheme
+
+        private init(_ value: _Scheme) {
+            self._scheme = value
+        }
+
+        /// LosslessStringConvertible` protocol requirements
+        public init?(_ description: String) {
+            guard let _scheme = _Scheme(rawValue: description) else {
+                return nil
+            }
+            self.init(_scheme)
+        }
+
+        public var description: String { self._scheme.rawValue }
+    }
+
+    /// Parses `input` into its constituent parts.
+    /// - Throws:
+    ///   - `MongoError.InvalidArgumentError` if the input is invalid.
+    public init(throwsIfInvalid input: String) throws {
+        let getScheme = input.components(separatedBy: "://")
+        guard getScheme.count > 1, let scheme = Scheme(getScheme[0]) else {
```

Comment:
We actually want to verify that the count is exactly 2, otherwise that means the user included multiple `://`, which I do not believe is allowed. We probably want a separate `guard` for that to give a more accurate error message though. Once we make  that assertion we can also assign a more meaningful variable name to `getScheme[1]`.

Reply:
done